File: leetcode/53.Maximum Subarray.cpp
/*
 * @lc app=leetcode.cn id=53 lang=cpp
 *
 * [53] 最大子序和
 */

// @lc code=start

#include <vector>

using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int result = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            nums[i] = max(nums[i], nums[i-1] + nums[i]);
            result = max(nums[i], result);
        }
        return result;
    }
};
// @lc code=end

File: leetcode/_53_Maximum_Subarray.java
/*
 * @lc app=leetcode.cn id=53 lang=java
 *
 * [53] 最大子序和
 */

// @lc code=start

public class _53_Maximum_Subarray {
    class Solution {
        public int maxSubArray(int[] nums) {
            if(nums.length == 0) {
                return Integer.MIN_VALUE;
            }
            int maxSum = nums[0];
            for (int i = 1; i < nums.length; i++) {
                nums[i] = Math.max(nums[i - 1] + nums[i], nums[i]);
                maxSum = Math.max(maxSum, nums[i]);
            }
            return maxSum;
        }
    }
}

// @lc code=end
